Lektura (#34)
infinum/frontend-handbook: JavaScript/Angular.md

userName$.subscribe(console.log); // TheDude
```

The reason why is because there might be multiple subscribers who want the same thing, and they would all have to do the same transformation in their respective success callbacks.

> It is, of course, possible that the subscribers want different things, and for those cases you might want to expose some observable whose operations are less piped.

Author: You're right, the meaning did change after proofreading, sorry, I thought you had something else in mind... Fewer piped operations is correct!

}
```

## Dependency injection

DI is utilized heavily in Angular, primarily for providing instances of either primitive values or class instances.

DI is a very complex topic and requires a separate discussion which would be out of the scope of this handbook. Make sure to check out the [official DI guide](https://angular.io/guide/dependency-injection) which explains many details. You can also have a look at [this repo](https://github.com/fvoska/angular-di-demo) for some examples.

As of Angular 6, make sure to use `{ providedIn: 'root' }` [for singletons](https://angular.io/guide/singleton-services#providing-a-singleton-service) whenever possible.

> DI is very useful for testing purposes, as shown in the next chapter.

Reviewer: I actually just noticed that I reordered some chapters after writing this line, so the chapter about testing is no longer immediately after DI chapter.

```suggestion
DI is very useful for testing purposes, as shown in a later chapter.
```

Author: Using "in a later chapter" is OK.
